### ISSUE TITLE
New-AzApiManagementApiSchema doesn't support OpenAPI

### ISSUE THREAD

**Issue by Splaxi** [Azure/azure-powershell#13247]
## Description
Working against a developer APIM instance, and trying to add a definition / schema for an API. The command doesn't fail, but the definition / schema isn't created at all. That is regardless of having the schema in a variable or using a schema file with the New-AzApiManagementApiSchema cmdlet.

I can create the definition / schema via portal.azure.com and it works like intended. Using the Get-AzApiManagementApiSchema does output some details, but the schema itself is missing (separate issue).

But using the API Management REST API, I can get and put the exact same schema for the API.

Using fiddler I can see there is a difference between the raw http put action done via invoke-restmethod and the http put comming from the New-AzApiManagementApiSchema.

## Steps to reproduce

Using the **New-AzApiManagementApiSchema** with a variable:
```powershell
Import-Module -Name Az.ApiManagement -Force
Connect-AzAccount -Subscription $subscriptionId -ErrorAction Stop
Set-AzContext -Subscription $subscriptionId -ErrorAction Stop
$context = New-AzApiManagementContext -ResourceGroupName $resourceGroup -ServiceName $apimServiceName

$schema = '{
    "components": {
        "schemas": {
            "Test": {
                "type": "object",
                "properties": {
                    "TestString": {
                        "type": "string"
                    },
                    "TestEmpty": {},
                    "TestArray": {
                        "type": "array",
                        "items": {
                            "required": [
                                "CreatedDate",
                                "CreatedBy_Name",
                                "CommentBody"
                            ],
                            "type": "object",
                            "properties": {
                                "CommentBody": {
                                    "type": "string"
                                },
                                "CreatedBy_Name": {},
                                "CreatedDate": {
                                    "type": "string"
                                }
                            }
                        }
                    },
                    "TestInteger": {
                        "type": "integer"
                    }
                }
            }
        }
    }
}'

$parms = @{
    ApiId                     = "foo-bar"
    SchemaDocument            = $schema
    SchemaId                  = $([System.Guid]::NewGuid().tostring())
    SchemaDocumentContentType = "OpenApiComponents"
}

New-AzApiManagementApiSchema -Context $context @parms
```

The fiddler trace for this call is below - sensitive and unrelated data removed:

```
PUT https://management.azure.com/subscriptions/2a*****20/resourceGroups/rg***dev/providers/Microsoft.ApiManagement/service/apim***-dev/apis/foo-bar/schemas/27014bcc-9971-46ec-801f-d30196fe83e5?api-version=2019-12-01 HTTP/1.1
x-ms-client-request-id: 08b6ffc7-061b-460f-975f-32748dff2fc8
accept-language: en-US
Authorization: Bearer eyJ*****aNl2PqQ
User-Agent: FxVersion/4.8.4250.0 OSName/Windows OSVersion/Microsoft.Windows.10.0.18363. Microsoft.Azure.Management.ApiManagement.ApiManagementClient/6.0.0.0 AzurePowershell/v0.0.0 PSVersion/v5.1.18362.1110
CommandName: New-AzApiManagementApiSchema
ParameterSetName: SchemaDocumentInline
Content-Type: application/json; charset=utf-8
Host: management.azure.com
Content-Length: 1554

{
  "properties": {
    "contentType": "application/vnd.oai.openapi.components+json",
    "document": {
      "value": "{\r\n    \"components\": {\r\n        \"schemas\": {\r\n            \"Test\": {\r\n                \"type\": \"object\",\r\n                \"properties\": {\r\n                    \"TestString\": {\r\n                        \"type\": \"string\"\r\n                    },\r\n                    \"TestEmpty\": {},\r\n                    \"TestArray\": {\r\n                        \"type\": \"array\",\r\n                        \"items\": {\r\n                            \"required\": [\r\n                                \"CreatedDate\",\r\n                                \"CreatedBy_Name\",\r\n                                \"CommentBody\"\r\n                            ],\r\n                            \"type\": \"object\",\r\n                            \"properties\": {\r\n                                \"CommentBody\": {\r\n                                    \"type\": \"string\"\r\n                                },\r\n                                \"CreatedBy_Name\": {},\r\n                                \"CreatedDate\": {\r\n                                    \"type\": \"string\"\r\n                                }\r\n                            }\r\n                        }\r\n                    },\r\n                    \"TestInteger\": {\r\n                        \"type\": \"integer\"\r\n                    }\r\n                }\r\n            }\r\n        }\r\n    }\r\n}"
    }
  }
}
```
The interesting part of above trace is the structure of the json. The document property has an object, with a value property. The content of that property is a **stringified json** object. Remember this when we compare the Api Management REST API call.

Using the Api Management REST API with a variable:
```powershell
$token = "eyJ*****aNl2PqQ"
$schema = '{
    "properties": {
        "contentType": "application/vnd.oai.openapi.components+json",
        "document": {
            "components": {
                "schemas": {
                    "Test": {
                        "type": "object",
                        "properties": {
                            "TestString": {
                                "type": "string"
                            },
                            "TestEmpty": {},
                            "TestArray": {
                                "type": "array",
                                "items": {
                                    "required": [
                                        "CreatedDate",
                                        "CreatedBy_Name",
                                        "CommentBody"
                                    ],
                                    "type": "object",
                                    "properties": {
                                        "CommentBody": {
                                            "type": "string"
                                        },
                                        "CreatedBy_Name": {},
                                        "CreatedDate": {
                                            "type": "string"
                                        }
                                    }
                                }
                            },
                            "TestInteger": {
                                "type": "integer"
                            }
                        }
                    }
                }
            }
        }
    }
}'

$uri = "https://management.azure.com/subscriptions/$subscriptionId/resourceGroups/$resourceGroup/providers/Microsoft.ApiManagement/service/$apimServiceName/apis/foo-bar/schemas/10217443-d8f5-44e5-b207-8ffe1ff2f2b7?api-version=2019-12-01"

# $response = Invoke-WebRequest -Uri 

Invoke-RestMethod -Method Put -Uri $uri -Headers @{Authorization = "Bearer $token"} -Body $schema 
```

The fiddler trace for this call is below - sensitive and unrelated data removed:
```
PUT https://management.azure.com/subscriptions/2a*****20/resourceGroups/rg***dev/providers/Microsoft.ApiManagement/service/apim***-dev/apis/foo-bar/schemas/10217443-d8f5-44e5-b207-8ffe1ff2f2b8?api-version=2019-12-01 HTTP/1.1
Authorization: Bearer eyJ*****aNl2PqQ
User-Agent: Mozilla/5.0 (Windows NT; Windows NT 10.0; da-DK) WindowsPowerShell/5.1.18362.1110
Host: management.azure.com
Content-Length: 1699
Expect: 100-continue

{
    "properties": {
        "contentType": "application/vnd.oai.openapi.components+json",
        "document": {
            "components": {
                "schemas": {
                    "Test": {
                        "type": "object",
                        "properties": {
                            "TestString": {
                                "type": "string"
                            },
                            "TestEmpty": {},
                            "TestArray": {
                                "type": "array",
                                "items": {
                                    "required": [
                                        "CreatedDate",
                                        "CreatedBy_Name",
                                        "CommentBody"
                                    ],
                                    "type": "object",
                                    "properties": {
                                        "CommentBody": {
                                            "type": "string"
                                        },
                                        "CreatedBy_Name": {},
                                        "CreatedDate": {
                                            "type": "string"
                                        }
                                    }
                                }
                            },
                            "TestInteger": {
                                "type": "integer"
                            }
                        }
                    }
                }
            }
        }
    }
}
```

The interesting part of above trace is the structure of the json. The document property has an object, with a components property. The content of this object is just plain json.

We have tried several different formats of the schema, to see if we had to many or to few properties. But when looking at the fiddler trace for when configuring the schema via portal.azure.com and the fact that we have a working schema when using the API Management REST API, we feel confident that there is an issue.

We have been digging into the code behind the command and believe the code that is creating this issue is:
https://github.com/Azure/azure-powershell/blob/aa0671e5f26ecfbf5a503af825691d2e247ce15b/src/ApiManagement/ApiManagement.ServiceManagement/ApiManagementClient.cs#L1370-L1374

We can see that there are several other issues around the API Management and OpenAPI/Swagger:
https://github.com/Azure/azure-api-management-devops-resource-kit/issues/264
https://github.com/Azure/azure-api-management-devops-resource-kit/issues/190
https://github.com/Azure/azure-powershell/issues/10626

## Environment data

```
Name                           Value
----                           -----
PSVersion                      5.1.18362.1110
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.1110
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

## Module versions

```powershell
PS C:\WINDOWS\system32> Get-Module  -ListAvailable                                                                      

    Directory: C:\Users\m******\Documents\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     4.6.1      Azure.Storage                       {Get-AzureStorageTable, New-AzureStorageTableSASToken, New...
Script     5.8.3      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...
Manifest   2.4.0.0    cChoco
Script     1.7.4.4    PoshRSJob                           {Get-RSJob, Receive-RSJob, Remove-RSJob, Start-RSJob...}
Script     1.0.19     PSFramework                         {Invoke-PSFProtectedCommand, Remove-PSFNull, Select-PSFObj...
Script     0.10.27... PSFramework                         {Remove-PSFNull, Select-PSFObject, Set-PSFConfig, Test-PSF...
Script     0.5.3      psnotification                      {Get-PSNUrl, Invoke-PSNHttpEndpoint, Invoke-PSNMessage, Se...


    Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Binary     1.4.118    APIManagementTemplate               {Get-ParameterTemplate, Write-APIManagementTemplates, Get-...
Binary     1.4.88     APIManagementTemplate               {Get-ParameterTemplate, Write-APIManagementTemplates, Get-...
Script     1.9.5      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     1.9.4      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     1.9.3      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     1.7.2      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     1.7.1      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     1.7.0      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     1.6.6      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     1.6.5      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     1.6.4      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     2.1.0      Az.ApiManagement                    {Add-AzApiManagementApiToGateway, Add-AzApiManagementApiTo...
Script     2.2.0      Az.KeyVault                         {Add-AzKeyVaultCertificate, Update-AzKeyVaultCertificate, ...
Script     1.3.2      Az.LogicApp                         {Get-AzIntegrationAccountAgreement, Get-AzIntegrationAccou...
Script     0.7.0      Az.Profile                          {Disable-AzDataCollection, Disable-AzContextAutosave, Enab...
Script     0.7.7      Az.ResourceGraph                    Search-AzGraph
Script     0.7.6      Az.ResourceGraph                    Search-AzGraph
Script     1.9.1      Az.Resources                        {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzR...
Script     1.9.0      Az.Resources                        {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzR...
Script     1.8.0      Az.Resources                        {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzR...
Script     1.7.1      Az.Resources                        {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzR...
Script     1.4.1      Az.ServiceBus                       {New-AzServiceBusNamespace, Get-AzServiceBusNamespace, Set...
Script     1.11.0     Az.Storage                          {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStor...
Script     1.10.0     Az.Storage                          {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStor...
Manifest   2.0.2      AzTable                             {Add-AzTableRow, Get-AzTableRow, Get-AzTableRowAll, Get-Az...
Script     0.5.4      Azure.AnalysisServices              {Add-AzureAnalysisServicesAccount, Restart-AzureAnalysisSe...
Script     4.6.1      Azure.Storage                       {Get-AzureStorageTable, New-AzureStorageTableSASToken, New...
Binary     2.0.2.76   AzureAD                             {Add-AzureADApplicationOwner, Get-AzureADApplication, Get-...
Binary     2.0.2.61   AzureAD                             {Add-AzureADApplicationOwner, Get-AzureADApplication, Get-...
Script     6.13.1     AzureRM
Script     0.6.14     AzureRM.AnalysisServices            {Resume-AzureRmAnalysisServicesServer, Suspend-AzureRmAnal...
Script     6.1.7      AzureRM.ApiManagement               {Add-AzureRmApiManagementRegion, Get-AzureRmApiManagementS...
Script     0.1.8      AzureRM.ApplicationInsights         {Get-AzureRmApplicationInsights, New-AzureRmApplicationIns...
Script     6.1.1      AzureRM.Automation                  {Get-AzureRMAutomationHybridWorkerGroup, Remove-AzureRmAut...
Script     4.0.11     AzureRM.Backup                      {Backup-AzureRmBackupItem, Enable-AzureRmBackupContainerRe...
Script     4.1.5      AzureRM.Batch                       {Remove-AzureRmBatchAccount, Get-AzureRmBatchAccount, Get-...
Script     0.14.6     AzureRM.Billing                     {Get-AzureRmBillingInvoice, Get-AzureRmBillingPeriod, Get-...
Script     5.0.6      AzureRM.Cdn                         {Get-AzureRmCdnProfile, Get-AzureRmCdnProfileSsoUrl, New-A...
Script     0.9.12     AzureRM.CognitiveServices           {Get-AzureRmCognitiveServicesAccount, Get-AzureRmCognitive...
Script     5.9.1      AzureRM.Compute                     {Remove-AzureRmAvailabilitySet, Get-AzureRmAvailabilitySet...
Script     0.3.7      AzureRM.Consumption                 {Get-AzureRmConsumptionBudget, Get-AzureRmConsumptionMarke...
Script     0.2.12     AzureRM.ContainerInstance           {New-AzureRmContainerGroup, Get-AzureRmContainerGroup, Rem...
Script     1.0.10     AzureRM.ContainerRegistry           {New-AzureRmContainerRegistry, Get-AzureRmContainerRegistr...
Script     5.0.3      AzureRM.DataFactories               {Remove-AzureRmDataFactory, Get-AzureRmDataFactoryRun, Get...
Script     0.5.11     AzureRM.DataFactoryV2               {Set-AzureRmDataFactoryV2, Update-AzureRmDataFactoryV2, Ge...
Script     5.1.4      AzureRM.DataLakeAnalytics           {Get-AzureRmDataLakeAnalyticsDataSource, New-AzureRmDataLa...
Script     6.2.1      AzureRM.DataLakeStore               {Get-AzureRmDataLakeStoreTrustedIdProvider, Remove-AzureRm...
Script     4.0.9      AzureRM.DevTestLabs                 {Get-AzureRmDtlAllowedVMSizesPolicy, Get-AzureRmDtlAutoShu...
Script     5.1.0      AzureRM.Dns                         {Get-AzureRmDnsRecordSet, New-AzureRmDnsRecordConfig, Remo...
Script     0.3.7      AzureRM.EventGrid                   {New-AzureRmEventGridTopic, Get-AzureRmEventGridTopic, Set...
Script     0.7.0      AzureRM.EventHub                    {New-AzureRmEventHubNamespace, Get-AzureRmEventHubNamespac...
Script     4.1.8      AzureRM.HDInsight                   {Get-AzureRmHDInsightJob, New-AzureRmHDInsightSqoopJobDefi...
Script     5.1.5      AzureRM.Insights                    {Get-AzureRmMetricDefinition, Get-AzureRmMetric, Remove-Az...
Script     3.1.8      AzureRM.IotHub                      {Add-AzureRmIotHubKey, Get-AzureRmIotHubEventHubConsumerGr...
Script     5.2.1      AzureRM.KeyVault                    {Add-AzureKeyVaultCertificate, Update-AzureKeyVaultCertifi...
Script     4.1.4      AzureRM.LogicApp                    {Get-AzureRmIntegrationAccountAgreement, Get-AzureRmIntegr...
Script     0.18.5     AzureRM.MachineLearning             {Move-AzureRmMlCommitmentAssociation, Get-AzureRmMlCommitm...
Script     0.4.8      AzureRM.MachineLearningCompute      {Get-AzureRmMlOpCluster, Get-AzureRmMlOpClusterKey, Test-A...
Script     0.2.7      AzureRM.MarketplaceOrdering         {Get-AzureRmMarketplaceTerms, Set-AzureRmMarketplaceTerms}
Script     0.10.4     AzureRM.Media                       {Sync-AzureRmMediaServiceStorageKeys, Set-AzureRmMediaServ...
Script     6.11.1     AzureRM.Network                     {Add-AzureRmApplicationGatewayAuthenticationCertificate, G...
Script     5.0.3      AzureRM.NotificationHubs            {Get-AzureRmNotificationHub, Get-AzureRmNotificationHubAut...
Script     5.0.6      AzureRM.OperationalInsights         {New-AzureRmOperationalInsightsAzureActivityLogDataSource,...
Script     1.1.0      AzureRM.PolicyInsights              {Get-AzureRmPolicyEvent, Get-AzureRmPolicyState, Get-Azure...
Script     4.1.10     AzureRM.PowerBIEmbedded             {Remove-AzureRmPowerBIWorkspaceCollection, Get-AzureRmPowe...
Script     5.8.3      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...
Script     5.8.2      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...
Script     4.1.9      AzureRM.RecoveryServices            {Get-AzureRmRecoveryServicesBackupProperty, Get-AzureRmRec...
Script     4.5.2      AzureRM.RecoveryServices.Backup     {Backup-AzureRmRecoveryServicesBackupItem, Get-AzureRmReco...
Script     0.2.12     AzureRM.RecoveryServices.SiteRec... {Edit-AzureRmRecoveryServicesAsrRecoveryPlan, Get-AzureRmR...
Script     5.1.0      AzureRM.RedisCache                  {Remove-AzureRmRedisCachePatchSchedule, New-AzureRmRedisCa...
Script     0.3.12     AzureRM.Relay                       {New-AzureRmRelayNamespace, Get-AzureRmRelayNamespace, Set...
Script     6.7.3      AzureRM.Resources                   {Get-AzureRmProviderOperation, Remove-AzureRmRoleAssignmen...
Script     0.16.10    AzureRM.Scheduler                   {Disable-AzureRmSchedulerJobCollection, Enable-AzureRmSche...
Script     0.6.13     AzureRM.ServiceBus                  {New-AzureRmServiceBusNamespace, Get-AzureRmServiceBusName...
Script     0.3.15     AzureRM.ServiceFabric               {Add-AzureRmServiceFabricApplicationCertificate, Add-Azure...
Script     1.0.0      AzureRM.SignalR                     {New-AzureRmSignalR, Get-AzureRmSignalR, Get-AzureRmSignal...
Script     4.12.1     AzureRM.Sql                         {Get-AzureRmSqlDatabaseTransparentDataEncryption, Get-Azur...
Script     5.2.0      AzureRM.Storage                     {Get-AzureRmStorageAccount, Get-AzureRmStorageAccountKey, ...
Script     4.0.10     AzureRM.StreamAnalytics             {Get-AzureRmStreamAnalyticsFunction, Get-AzureRmStreamAnal...
Script     4.0.5      AzureRM.Tags                        {Remove-AzureRmTag, Get-AzureRmTag, New-AzureRmTag}
Script     4.1.3      AzureRM.TrafficManager              {Add-AzureRmTrafficManagerCustomHeaderToEndpoint, Remove-A...
Script     4.0.5      AzureRM.UsageAggregates             Get-UsageAggregates
Script     5.2.0      AzureRM.Websites                    {Get-AzureRmAppServicePlan, Set-AzureRmAppServicePlan, New...
Script     3.5.1.458  CosmosDB                            {Get-CosmosDbAccount, Get-CosmosDbAccountConnectionString,...
Script     0.2.2      d365ce.integrations                 {Add-D365CeODataConfig, Enable-D365CeException, Get-D365Ce...
Script     0.4.14     d365fo.integrations                 {Add-D365ODataConfig, Enable-D365ExceptionIntegrations, Ex...
Script     0.4.13     d365fo.integrations                 {Add-D365ODataConfig, Enable-D365ExceptionIntegrations, Ex...
Script     0.4.11     d365fo.integrations                 {Add-D365ODataConfig, Enable-D365ExceptionIntegrations, Ex...
Script     0.4.10     d365fo.integrations                 {Add-D365ODataConfig, Enable-D365ExceptionIntegrations, Ex...
Script     0.3.3      d365fo.integrations                 {Add-D365ODataConfig, Enable-d365Exception, Export-D365Dmf...
Script     0.6.31     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     0.6.25     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     0.6.20     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     0.6.19     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     0.6.11     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     0.5.81     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     0.5.79     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     0.5.74     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     0.5.64     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     0.5.55     d365fo.tools                        {Add-D365AzureStorageConfig, Add-D365BroadcastMessageConfi...
Script     1.0.72     dbatools                            {Select-DbaObject, Set-DbatoolsConfig, Start-DbaMigration,...
Script     1.0.1      ExchangeOnlineManagement            {Get-EXOCasMailbox, Get-EXOMailbox, Get-EXOMailboxFolderPe...
Script     7.1.1      ImportExcel                         {Add-ConditionalFormatting, Add-ExcelChart, Add-ExcelDataV...
Script     7.1.0      ImportExcel                         {Add-ConditionalFormatting, Add-ExcelChart, Add-ExcelDataV...
Script     7.0.1      ImportExcel                         {Add-ConditionalFormatting, Add-ExcelChart, Add-ExcelDataV...
Script     6.5.2      ImportExcel                         {Add-ConditionalFormatting, Add-ExcelChart, Add-ExcelDataV...
Script     5.5.6      InvokeBuild                         {Invoke-Build, Build-Checkpoint, Build-Parallel}
Binary     1.0.23     LogicAppTemplate                    {Get-EmptyParameterTemplate, Get-CustomConnectorTemplate, ...
Binary     1.0.20     LogicAppTemplate                    {Get-EmptyParameterTemplate, Get-CustomConnectorTemplate, ...
Binary     1.0.14     LogicAppTemplate                    {Get-EmptyParameterTemplate, Get-CustomConnectorTemplate, ...
Binary     1.0.12     LogicAppTemplate                    {Get-EmptyParameterTemplate, Get-CustomConnectorTemplate, ...
Script     1.0.1      Microsoft.PowerShell.Operation.V... {Get-OperationValidation, Invoke-OperationValidation}
Script     2.8.8      Microsoft.Xrm.Data.Powershell       {Get-CrmConnection, Get-CrmOrganizations, Invoke-CrmAction...
Manifest   1.1.183.57 MSOnline                            {Get-MsolDevice, Remove-MsolDevice, Enable-MsolDevice, Dis...
Script     1.0.2.201  newtonsoft.json                     {ConvertFrom-JsonNewtonsoft, ConvertTo-JsonNewtonsoft}
Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-Packa...
Script     4.10.0     Pester                              {Describe, Context, It, Should...}
Script     4.9.0      Pester                              {Describe, Context, It, Should...}
Script     3.4.0      Pester                              {Describe, Context, It, Should...}
Script     0.14.0     platyPS                             {New-MarkdownHelp, Get-MarkdownMetadata, New-ExternalHelp,...
Script     1.7.4.4    PoshRSJob                           {Get-RSJob, Receive-RSJob, Remove-RSJob, Start-RSJob...}
Script     1.0.0.1    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}
Script     0.2.8      PSDevOps                            {Remove-ADOProject, Set-ADOWorkItem, Get-ADOBuild, Remove-...
Script     1.4.150    PSFramework                         {ConvertTo-PSFHashtable, Invoke-PSFCallback, Invoke-PSFPro...
Script     1.4.149    PSFramework                         {ConvertTo-PSFHashtable, Invoke-PSFCallback, Invoke-PSFPro...
Script     1.1.59     PSFramework                         {ConvertTo-PSFHashtable, Invoke-PSFCallback, Invoke-PSFPro...
Script     2.2.7.98   PSModuleDevelopment                 {Expand-PSMDTypeName, Export-PSMDString, Find-PSMDFileCont...
Script     2.2.7.90   PSModuleDevelopment                 {Expand-PSMDTypeName, Export-PSMDString, Find-PSMDFileCont...
Script     0.5.3      psnotification                      {Get-PSNUrl, Invoke-PSNHttpEndpoint, Invoke-PSNMessage, Se...
Script     0.3.0      PSOAuthHelper                       {Get-BearerToken, Get-RemainingMinutes, Invoke-AzureResour...
Script     0.2.6      PSOAuthHelper                       {Get-BearerToken, Get-RemainingMinutes, Invoke-AzureResour...
Script     2.0.0      PSReadline                          {Get-PSReadLineKeyHandler, Set-PSReadLineKeyHandler, Remov...
Script     1.18.3     PSScriptAnalyzer                    {Get-ScriptAnalyzerRule, Invoke-ScriptAnalyzer, Invoke-For...
Script     1.2.0      PSServiceBus                        {Clear-SbQueue, Get-SbQueue, Get-SbTopic, Get-SbSubscripti...
Script     1.0.1      PSServiceBus                        {Get-SbQueue, Get-SbTopic, Get-SbSubscription, Send-SbMess...
Script     1.0        RemoteDesktopManager.PowerShellM...


    Directory: C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Manifest   1.0.0.0    AppBackgroundTask                   {Disable-AppBackgroundTaskDiagnosticLog, Enable-AppBackgro...
Manifest   2.0.0.0    AppLocker                           {Get-AppLockerFileInformation, Get-AppLockerPolicy, New-Ap...
Manifest   1.0.0.0    AppvClient                          {Add-AppvClientConnectionGroup, Add-AppvClientPackage, Add...
Manifest   2.0.1.0    Appx                                {Add-AppxPackage, Get-AppxPackage, Get-AppxPackageManifest...
Script     1.0.0.0    AssignedAccess                      {Clear-AssignedAccess, Get-AssignedAccess, Set-AssignedAcc...
Manifest   1.0.0.0    BitLocker                           {Unlock-BitLocker, Suspend-BitLocker, Resume-BitLocker, Re...
Manifest   2.0.0.0    BitsTransfer                        {Add-BitsFile, Complete-BitsTransfer, Get-BitsTransfer, Re...
Manifest   1.0.0.0    BranchCache                         {Add-BCDataCacheExtension, Clear-BCCache, Disable-BC, Disa...
Manifest   1.0.0.0    CimCmdlets                          {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance,...
Manifest   1.0        ConfigCI                            {Get-SystemDriver, New-CIPolicyRule, New-CIPolicy, Get-CIP...
Manifest   1.0        Defender                            {Get-MpPreference, Set-MpPreference, Add-MpPreference, Rem...
Manifest   1.0.2.0    DeliveryOptimization                {Delete-DeliveryOptimizationCache, Get-DeliveryOptimizatio...
Manifest   1.0.0.0    DirectAccessClientComponents        {Disable-DAManualEntryPointSelection, Enable-DAManualEntry...
Script     3.0        Dism                                {Add-AppxProvisionedPackage, Add-WindowsDriver, Add-Window...
Manifest   1.0.0.0    DnsClient                           {Resolve-DnsName, Clear-DnsClientCache, Get-DnsClient, Get...
Manifest   1.0.0.0    EventTracingManagement              {Start-EtwTraceSession, New-EtwTraceSession, Get-EtwTraceS...
Manifest   2.0.0.0    International                       {Get-WinDefaultInputMethodOverride, Set-WinDefaultInputMet...
Manifest   1.0.0.0    iSCSI                               {Get-IscsiTargetPortal, New-IscsiTargetPortal, Remove-Iscs...
Script     1.0.0.0    ISE                                 {New-IseSnippet, Import-IseSnippet, Get-IseSnippet}
Manifest   1.0.0.0    Kds                                 {Add-KdsRootKey, Get-KdsRootKey, Test-KdsRootKey, Set-KdsC...
Manifest   1.0.1.0    Microsoft.PowerShell.Archive        {Compress-Archive, Expand-Archive}
Manifest   3.0.0.0    Microsoft.PowerShell.Diagnostics    {Get-WinEvent, Get-Counter, Import-Counter, Export-Counter...
Manifest   3.0.0.0    Microsoft.PowerShell.Host           {Start-Transcript, Stop-Transcript}
Manifest   1.0.0.0    Microsoft.PowerShell.LocalAccounts  {Add-LocalGroupMember, Disable-LocalUser, Enable-LocalUser...
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-ItemProperty, Join-Path...
Script     1.0        Microsoft.PowerShell.ODataUtils     Export-ODataEndpointProxy
Manifest   3.0.0.0    Microsoft.PowerShell.Security       {Get-Acl, Set-Acl, Get-PfxCertificate, Get-Credential...}
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Format-List, Format-Custom, Format-Table, Format-Wide...}
Manifest   3.0.0.0    Microsoft.WSMan.Management          {Disable-WSManCredSSP, Enable-WSManCredSSP, Get-WSManCredS...
Manifest   1.0        MMAgent                             {Disable-MMAgent, Enable-MMAgent, Set-MMAgent, Get-MMAgent...
Manifest   1.0.0.0    MsDtc                               {New-DtcDiagnosticTransaction, Complete-DtcDiagnosticTrans...
Manifest   2.0.0.0    NetAdapter                          {Disable-NetAdapter, Disable-NetAdapterBinding, Disable-Ne...
Manifest   1.0.0.0    NetConnection                       {Get-NetConnectionProfile, Set-NetConnectionProfile}
Manifest   1.0.0.0    NetDiagnostics                      Get-NetView
Manifest   1.0.0.0    NetEventPacketCapture               {New-NetEventSession, Remove-NetEventSession, Get-NetEvent...
Manifest   2.0.0.0    NetLbfo                             {Add-NetLbfoTeamMember, Add-NetLbfoTeamNic, Get-NetLbfoTea...
Manifest   1.0.0.0    NetNat                              {Get-NetNat, Get-NetNatExternalAddress, Get-NetNatStaticMa...
Manifest   2.0.0.0    NetQos                              {Get-NetQosPolicy, Set-NetQosPolicy, Remove-NetQosPolicy, ...
Manifest   2.0.0.0    NetSecurity                         {Get-DAPolicyChange, New-NetIPsecAuthProposal, New-NetIPse...
Manifest   1.0.0.0    NetSwitchTeam                       {New-NetSwitchTeam, Remove-NetSwitchTeam, Get-NetSwitchTea...
Manifest   1.0.0.0    NetTCPIP                            {Get-NetIPAddress, Get-NetIPInterface, Get-NetIPv4Protocol...
Manifest   1.0.0.0    NetworkConnectivityStatus           {Get-DAConnectionStatus, Get-NCSIPolicyConfiguration, Rese...
Manifest   1.0.0.0    NetworkSwitchManager                {Disable-NetworkSwitchEthernetPort, Enable-NetworkSwitchEt...
Manifest   1.0.0.0    NetworkTransition                   {Add-NetIPHttpsCertBinding, Disable-NetDnsTransitionConfig...
Manifest   1.0.0.0    PcsvDevice                          {Get-PcsvDevice, Start-PcsvDevice, Stop-PcsvDevice, Restar...
Binary     1.0.0.0    PersistentMemory                    {Get-PmemDisk, Get-PmemPhysicalDevice, Get-PmemUnusedRegio...
Manifest   1.0.0.0    PKI                                 {Add-CertificateEnrollmentPolicyServer, Export-Certificate...
Manifest   1.0.0.0    PnpDevice                           {Get-PnpDevice, Get-PnpDeviceProperty, Enable-PnpDevice, D...
Manifest   1.1        PrintManagement                     {Add-Printer, Add-PrinterDriver, Add-PrinterPort, Get-Prin...
Binary     1.0.11     ProcessMitigations                  {Get-ProcessMitigation, Set-ProcessMitigation, ConvertTo-P...
Script     3.0        Provisioning                        {Install-ProvisioningPackage, Export-ProvisioningPackage, ...
Manifest   1.1        PSDesiredStateConfiguration         {Set-DscLocalConfigurationManager, Start-DscConfiguration,...
Script     1.0.0.0    PSDiagnostics                       {Disable-PSTrace, Disable-PSWSManCombinedTrace, Disable-WS...
Binary     1.1.0.0    PSScheduledJob                      {New-JobTrigger, Add-JobTrigger, Remove-JobTrigger, Get-Jo...
Manifest   2.0.0.0    PSWorkflow                          {New-PSWorkflowExecutionOption, New-PSWorkflowSession, nwsn}
Manifest   1.0.0.0    PSWorkflowUtility                   Invoke-AsWorkflow
Manifest   1.0.0.0    ScheduledTasks                      {Get-ScheduledTask, Set-ScheduledTask, Register-ScheduledT...
Manifest   2.0.0.0    SecureBoot                          {Confirm-SecureBootUEFI, Set-SecureBootUEFI, Get-SecureBoo...
Manifest   2.0.0.0    SmbShare                            {Get-SmbShare, Remove-SmbShare, Set-SmbShare, Block-SmbSha...
Manifest   2.0.0.0    SmbWitness                          {Get-SmbWitnessClient, Move-SmbWitnessClient, gsmbw, msmbw...
Manifest   1.0.0.0    StartLayout                         {Export-StartLayout, Import-StartLayout, Export-StartLayou...
Manifest   2.0.0.0    Storage                             {Add-InitiatorIdToMaskingSet, Add-PartitionAccessPath, Add...
Manifest   1.0.0.0    StorageBusCache                     {Clear-StorageBusDisk, Disable-StorageBusCache, Disable-St...
Manifest   2.0.0.0    TLS                                 {New-TlsSessionTicketKey, Enable-TlsSessionTicketKey, Disa...
Manifest   1.0.0.0    TroubleshootingPack                 {Get-TroubleshootingPack, Invoke-TroubleshootingPack}
Manifest   2.0.0.0    TrustedPlatformModule               {Get-Tpm, Initialize-Tpm, Clear-Tpm, Unblock-Tpm...}
Binary     2.1.639.0  UEV                                 {Clear-UevConfiguration, Clear-UevAppxPackage, Restore-Uev...
Manifest   2.0.0.0    VpnClient                           {Add-VpnConnection, Set-VpnConnection, Remove-VpnConnectio...
Manifest   1.0.0.0    Wdac                                {Get-OdbcDriver, Set-OdbcDriver, Get-OdbcDsn, Add-OdbcDsn...}
Manifest   2.0.0.0    Whea                                {Get-WheaMemoryPolicy, Set-WheaMemoryPolicy}
Manifest   1.0.0.0    WindowsDeveloperLicense             {Get-WindowsDeveloperLicense, Unregister-WindowsDeveloperL...
Script     1.0        WindowsErrorReporting               {Enable-WindowsErrorReporting, Disable-WindowsErrorReporti...
Manifest   1.0.0.0    WindowsSearch                       {Get-WindowsSearchSetting, Set-WindowsSearchSetting}
Manifest   1.0.0.0    WindowsUpdate                       Get-WindowsUpdateLog
Manifest   1.0.0.2    WindowsUpdateProvider               {Get-WUAVersion, Get-WULastInstallationDate, Get-WULastSca...
```

## Debug output

Private and sensitive data has been masked
```
DEBUG: 20:26:44 - GetAzureRMContextCommand begin processing with ParameterSet 'GetSingleContext'.
DEBUG: 20:26:44 - using account id 'xyz@test.com'...
DEBUG: AzureQoSEvent: CommandName - Get-AzContext; IsSuccess - True; Duration - 00:00:00.0533102;
DEBUG: Finish sending metric.
DEBUG: 20:26:46 - GetAzureRMContextCommand end processing.
DEBUG: 20:26:46 - GetAzureRMContextCommand begin processing with ParameterSet 'GetSingleContext'.
DEBUG: 20:26:46 - using account id 'xyz@test.com'...
DEBUG: AzureQoSEvent: CommandName - Get-AzContext; IsSuccess - True; Duration - 00:00:00.0032923;
DEBUG: Finish sending metric.
DEBUG: 20:26:46 - GetAzureRMContextCommand end processing.
DEBUG: 20:26:46 - NewAzureApiManagementContext begin processing with ParameterSet 'ContextParameterSet'.
DEBUG: 20:26:46 - using account id 'xyz@test.com'...
DEBUG: AzureQoSEvent: CommandName - New-AzApiManagementContext; IsSuccess - True; Duration - 00:00:00.0034128;
DEBUG: Finish sending metric.
DEBUG: 20:26:47 - NewAzureApiManagementContext end processing.
DEBUG: 20:26:47 - NewAzureApiManagementApiSchema begin processing with ParameterSet 'SchemaDocumentInline'.
DEBUG: 20:26:47 - using account id 'xyz@test.com'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'xyz@test.com', environment: 'AzureCloud', tenant:
'50***********be'
DEBUG: [Common.Authentication]: Authenticating using configuration values: Domain:
'50***********be', Endpoint: 'https://login.microsoftonline.com/', ClientId:
'19******************c2', ClientRedirect: 'urn:ietf:wg:oauth:2.0:oob', ResourceClientUri:
'https://management.core.windows.net/', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using context with Authority
'https://login.microsoftonline.com/50***********be/', CorrelationId:
'00000000-0000-0000-0000-000000000000', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using AdalConfiguration with Domain:
'50***********be', AdEndpoint: 'https://login.microsoftonline.com/', ClientId:
'19******************c2', ClientRedirectUri: urn:ietf:wg:oauth:2.0:oob
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4632435Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: ADAL
PCL.Desktop with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version
'2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4632435Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: ADAL
PCL.Desktop with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version
'2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4632435Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: ===
Token Acquisition started:
 CacheType: null
 Authentication Target: User
 , Authority Host: login.microsoftonline.com
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4632435Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: ===
Token Acquisition started:
 Authority: https://login.microsoftonline.com/50***********be/
 Resource: https://management.core.windows.net/
 ClientId: 19******************c2
 CacheType: null
 Authentication Target: User

DEBUG: [ADAL]: Verbose: 2020-10-16T18:26:47.4632435Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: Loading
from cache.
DEBUG: [ADAL]: Verbose: 2020-10-16T18:26:47.4632435Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: Loading
from cache.
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4652363Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs:
Deserialized 22 items to token cache.
DEBUG: [ADAL]: Verbose: 2020-10-16T18:26:47.4652363Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: Looking up
cache for a token...
DEBUG: [ADAL]: Verbose: 2020-10-16T18:26:47.4662361Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: Looking up
cache for a token...
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4662361Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: An item
 matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4662361Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: An item
 matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4662361Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs:
24,4301293983333 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4662361Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs:
24,4301293983333 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4662361Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: A
matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4662361Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: A
matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4662361Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: ===
Token Acquisition finished successfully. An access token was returned: Expiration Time: 16-10-2020 18:51:13 +00:00
DEBUG: [ADAL]: Information: 2020-10-16T18:26:47.4662361Z: db036e64-7868-48e1-8a67-9179e22027a0 - LoggerBase.cs: ===
Token Acquisition finished successfully. An access token was returned: Expiration Time: 16-10-2020 18:51:13
+00:00Access Token Hash: t8*********************=
  User id: 5e******************db
DEBUG: [Common.Authentication]: Renewing Token with Type: 'Bearer', Expiry: '10/16/2020 18:51:13 +00:00',
MultipleResource? 'True', Tenant: '50***********be', UserId: 'xyz@test.com'
DEBUG: [Common.Authentication]: User info for token DisplayId: 'xyz@test.com', Name: M********, IdProvider:
'https://sts.windows.net/50***********be/', Uid: '5e******************db'
DEBUG: [Common.Authentication]: Checking token expiration, token expires '10/16/2020 18:51:13 +00:00' Comparing to
'10/16/2020 18:26:47 +00:00' With threshold '00:05:00', calculated time until token expiry: '00:24:25.8038104'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/2a*****************20/resourceGroups/rg**************-dev/p
roviders/Microsoft.ApiManagement/service/apim**************-dev/apis/foo-bar/schemas/1b9d3379-d598-4d26-8fcb-a9989d45
5e98?api-version=2019-12-01

Headers:
x-ms-client-request-id        : 199f4017-50f9-4f05-ace2-3486a0587f19
accept-language               : en-US

Body:
{
  "properties": {
    "contentType": "application/vnd.oai.openapi.components+json",
    "document": {
      "value": "\"components\": {\r\n    \"schemas\": {\r\n        \"Test\": {\r\n            \"type\": \"object\",\r\n
            \"properties\": {\r\n                \"TestString\": {\r\n                    \"type\": \"string\"\r\n
           },\r\n                \"TestEmpty\": {},\r\n                \"TestArray\": {\r\n
\"type\": \"array\",\r\n                    \"items\": {\r\n                        \"required\": [\r\n
            \"CreatedDate\",\r\n                            \"CreatedBy_Name\",\r\n
\"CommentBody\"\r\n                        ],\r\n                        \"type\": \"object\",\r\n
   \"properties\": {\r\n                            \"CommentBody\": {\r\n                                \"type\":
\"string\"\r\n                            },\r\n                            \"CreatedBy_Name\": {},\r\n
            \"CreatedDate\": {\r\n                                \"type\": \"string\"\r\n
}\r\n                        }\r\n                    }\r\n                },\r\n                \"TestInteger\": {\r\n
                    \"type\": \"integer\"\r\n                }\r\n            }\r\n        }\r\n    }\r\n}"
    }
  }
}

DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
Accepted

Headers:
Pragma                        : no-cache
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-request-id               : 8776b4f5-f065-404c-a83b-e0318c1551da
x-ms-ratelimit-remaining-subscription-writes: 1199
x-ms-correlation-request-id   : 6549f1a0-818d-436d-b185-4d2cf8a7a94f
x-ms-routing-request-id       : GERMANYNORTH:20201016T182649Z:6549f1a0-818d-436d-b185-4d2cf8a7a94f
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Fri, 16 Oct 2020 18:26:48 GMT
Location                      :
https://management.azure.com/subscriptions/2a*****************20/resourceGroups/rg**************-dev/p
roviders/Microsoft.ApiManagement/service/apim**************-dev/apis/foo-bar/schemas/1b9d3379-d598-4d26-8fcb-a9989d45
5e98?api-version=2019-12-01&asyncId=5f89e5e96fc0f406e063f7d9&asyncCode=201
Server                        : Microsoft-HTTPAPI/2.0

Body:


DEBUG: [Common.Authentication]: Renewing Token with Type: 'Bearer', Expiry: '10/16/2020 18:51:13 +00:00',
MultipleResource? 'True', Tenant: '50***********be', UserId: 'xyz@test.com'
DEBUG: [Common.Authentication]: User info for token DisplayId: 'xyz@test.com', Name: M********, IdProvider:
'https://sts.windows.net/50***********be/', Uid: '5e******************db'
DEBUG: [Common.Authentication]: Checking token expiration, token expires '10/16/2020 18:51:13 +00:00' Comparing to
'10/16/2020 18:27:19 +00:00' With threshold '00:05:00', calculated time until token expiry: '00:23:53.9698660'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/subscriptions/2a*****************20/resourceGroups/rg**************-dev/p
roviders/Microsoft.ApiManagement/service/apim**************-dev/apis/foo-bar/schemas/1b9d3379-d598-4d26-8fcb-a9989d45
5e98?api-version=2019-12-01&asyncId=5f89e5e96fc0f406e063f7d9&asyncCode=201

Headers:

Body:


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
Created

Headers:
Pragma                        : no-cache
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-request-id               : 0160bc1b-f35e-41a9-b118-f267cfb21518
x-ms-ratelimit-remaining-subscription-reads: 11998
x-ms-correlation-request-id   : 980c52eb-527f-419e-a931-c8b89a8de09c
x-ms-routing-request-id       : GERMANYNORTH:20201016T182719Z:980c52eb-527f-419e-a931-c8b89a8de09c
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Fri, 16 Oct 2020 18:27:19 GMT
ETag                          : "AAAAAAAACn0="
Server                        : Microsoft-HTTPAPI/2.0

Body:
{
  "id":
"/subscriptions/2a*****************20/resourceGroups/rg**************-dev/providers/Microsoft.ApiManag
ement/service/apim**************-dev/apis/foo-bar/schemas/1b9d3379-d598-4d26-8fcb-a9989d455e98",
  "type": "Microsoft.ApiManagement/service/apis/schemas",
  "name": "1b9d3379-d598-4d26-8fcb-a9989d455e98",
  "properties": {
    "contentType": "application/vnd.oai.openapi.components+json",
    "document": {
      "value": "\"components\": {\r\n    \"schemas\": {\r\n        \"Test\": {\r\n            \"type\": \"object\",\r\n
            \"properties\": {\r\n                \"TestString\": {\r\n                    \"type\": \"string\"\r\n
           },\r\n                \"TestEmpty\": {},\r\n                \"TestArray\": {\r\n
\"type\": \"array\",\r\n                    \"items\": {\r\n                        \"required\": [\r\n
            \"CreatedDate\",\r\n                            \"CreatedBy_Name\",\r\n
\"CommentBody\"\r\n                        ],\r\n                        \"type\": \"object\",\r\n
   \"properties\": {\r\n                            \"CommentBody\": {\r\n                                \"type\":
\"string\"\r\n                            },\r\n                            \"CreatedBy_Name\": {},\r\n
            \"CreatedDate\": {\r\n                                \"type\": \"string\"\r\n
}\r\n                        }\r\n                    }\r\n                },\r\n                \"TestInteger\": {\r\n
                    \"type\": \"integer\"\r\n                }\r\n            }\r\n        }\r\n    }\r\n}"
    }
  }
}



SchemaId           : 1b9d3379-d598-4d26-8fcb-a9989d455e98
Api Id             : foo-bar
Schema ContentType : OpenApiComponents
Schema Document    : "components": {
                         "schemas": {
                          ....

DEBUG: AzureQoSEvent: CommandName - New-AzApiManagementApiSchema; IsSuccess - True; Duration - 00:00:32.2310338;
DEBUG: Finish sending metric.
DEBUG: 20:27:21 - NewAzureApiManagementApiSchema end processing.
```

## Error output

```
DEBUG: 20:36:27 - ResolveError begin processing with ParameterSet 'AnyErrorParameterSet'.
DEBUG: 20:36:27 - using account id 'xyz@test.com'...
WARNING: Upcoming breaking changes in the cmdlet 'Resolve-AzError' :
The `Resolve-Error` alias will be removed in a future release.  Please change any scripts that use this alias to use
`Resolve-AzError` instead.
Note : Go to https://aka.ms/azps-changewarnings for steps to suppress this breaking change warning, and other
information on breaking changes in Azure PowerShell.

The Azure PowerShell team is listening, please let us know how we are doing: https://aka.ms/azpssurvey?Q_CHL=ERROR.

DEBUG: AzureQoSEvent: CommandName - Resolve-AzError; IsSuccess - True; Duration - 00:00:00.0288439;
DEBUG: Finish sending metric.
DEBUG: 20:36:28 - ResolveError end processing.
```

**Comment by miaojiang** [comment]
cc @solankisamir

**Comment by dingmeng-xue** [comment]
Service team, please help to look into this question.

**Comment by Splaxi** [comment]
Could I get an update on the issue?

**Comment by miaojiang** [comment]
@solankisamir could you please take a look?

**Comment by Splaxi** [comment]
Any updates on this issue?

**Comment by omkarg81** [comment]
@solankisamir @miaojiang Any updates on this issue? I was trying to use the said REST API call, as equivalent powershell cmdlet doesn't function as expected for adding schema-definitions. But the problem is even with REST API call, it is able to add one definition at a time. If my schema has more than 1 definition, when I fire the 2nd and subsequent PUT call to https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/apis/{apiName}/schemas/{schemaId}?api-version=2021-01-01-preview

it overwrites the previously written definition, and finally only 1 definition remains. Tried adding the If-Match to Request Header on 2nd and subsequent calls too, but in vain.

Tried adding multiple definitions under "schemas" as array of json, but even if that creates multiple definitions in 1 go, the DefinitionName are 0, 1, 2, 3 etc. and not actual names given in the input json body.

Multiple Definition Sample below - 
`{
  "properties": {
    "contentType": "application/vnd.oai.openapi.components+json",
    "document": {
      "components": {
        "schemas": **[
          {
            "### Definition1": {
              "type": "object",
              "properties": {
                "String1": {
                  "type": "string"
                }
              }
            },
            "### Definition2": {
              "type": "object",
              "properties": {
                "String2": {
                  "type": "integer"
                }
              }
            }
          }
        ]**
      }
    }
  }
}`

Does the PUT call allow putting definitions at once and if so, how?
